unit_tests/src/test_replica.cpp

```diff
@@ -0,0 +1,87 @@
+#include "catch.hpp"
+
+#include "connection_pool.hpp"
+#include "dstream.hpp"
+#include "irods_at_scope_exit.hpp"
+#include "replica.hpp"
+#include "rodsClient.h"
+#include "transport/default_transport.hpp"
+
+#include <chrono>
+#include <iostream>
+#include <string_view>
+#include <thread>
+
+namespace fs = irods::experimental::filesystem;
+namespace io = irods::experimental::io;
+namespace replica = irods::experimental::replica;
+
+TEST_CASE("replica", "[replica]")
+{
+    load_client_api_plugins();
+
+    auto conn_pool = irods::make_connection_pool();
+    auto conn = static_cast<rcComm_t>(conn_pool->get_connection());
```

Comment:
this results in a copy (i think).  im not entirely sure what this means for the connection proxy and pool.

change to `rcComm_t&`